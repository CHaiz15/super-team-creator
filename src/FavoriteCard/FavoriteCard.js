import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { deleteFavorite } from '../Actions'
import './FavoriteCard.scss'

export const FavoriteCard = ({team, deleteFavorite}) => {

  return (
    <>
      <Link to={{pathname:'/favTeam', state: {team}}}> 
        <div className='fav-card'>
          <h2>{team.name}</h2>
          <img className='member-image' src={team.memberOne.image.url} alt='member one'/>
          <img className='member-image' src={team.memberTwo.image.url} alt='member two'/>
          <img className='member-image' src={team.memberThree.image.url} alt='member three'/>
        </div>
      </Link>
      <button onClick={() => deleteFavorite(team.id)}>Delete</button>
    </>
  )
}

export const mapDispatchToProps = dispatch => ({
  deleteFavorite: id => dispatch(deleteFavorite(id)),
})

export default connect(null, mapDispatchToProps)(FavoriteCard)