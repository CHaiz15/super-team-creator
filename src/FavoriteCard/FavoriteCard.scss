.fav-card {
  cursor: pointer;
  height: 10%;
  border: 2px solid black;
}