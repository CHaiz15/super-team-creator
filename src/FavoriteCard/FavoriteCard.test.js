import React from 'react';
import { FavoriteCard } from './FavoriteCard.js';
import { shallow } from 'enzyme'

describe('FavoriteCard', () => {
  let wrapper;
  let mockTeam;
  beforeEach(() => {
      mockTeam = {
        name:'Best Team Ever',
        memberOne: {
          name:'Hulk', 
          powerstats: {
            Intelligence: 81,
            strength: 40,
            Speed: 29,
            Durability: 55,
            Power: 63,
            Combat: 90,
          },
          image: {url: "https://www.superherodb.com/pictures2/portraits/10/100/83.jpg"},
        },
        memberTwo: {
          name:'Thanos',
          powerstats: {
            Intelligence: 81,
            strength: 40,
            Speed: 29,
            Durability: 55,
            Power: 63,
            Combat: 90,
          },
          image: {url: "https://www.superherodb.com/pictures2/portraits/10/100/83.jpg"},
        },
        memberThree: {
          name:'Batman',
          powerstats: {
            Intelligence: 81,
            strength: 40,
            Speed: 29,
            Durability: 55,
            Power: 63,
            Combat: 90,
          },
          image: {url: "https://www.superherodb.com/pictures2/portraits/10/100/83.jpg"},
        },
        id: 123,
      }
    wrapper = shallow(<FavoriteCard key={mockTeam.id} team={mockTeam} deleteFavorite={jest.fn()}/>);
  })  
  it('should be an instance of the FavoriteCard component', () => {
      expect(wrapper).toMatchSnapshot();
  });
})
