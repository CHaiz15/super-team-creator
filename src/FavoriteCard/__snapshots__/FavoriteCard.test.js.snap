// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteCard should be an instance of the FavoriteCard component 1`] = `ShallowWrapper {}`;
